src/sketch.lisp

```diff
@@ -106,13 +117,30 @@
 
 (defmethod initialize-instance :around ((instance sketch) &key &allow-other-keys)
   (initialize-sketch)
-  (call-next-method)
+  (sdl2:in-main-thread ()
+    (call-next-method))
   (kit.sdl2:start))
 
 (defmethod initialize-instance :after ((instance sketch) &rest initargs &key &allow-other-keys)
-  (initialize-environment instance)
   (apply #'prepare instance initargs)
-  (initialize-gl instance))
+  (setf (sketch-%window instance)
+        (make-instance 'sketch-window
+                       :title (sketch-title instance)
+                       :w (sketch-width instance)
+                       :h (sketch-height instance)
+                       :fullscreen (sketch-fullscreen instance)
+                       :resizable (sketch-resizable instance)
+                       :sketch instance))
+  (initialize-environment instance)
+  (initialize-gl instance)
+  ;; These will have been added in the call to PREPARE.
+  (with-slots ((fs %delayed-init-funs)) instance
+    (loop for f across fs
+          do (funcall f))
+    ;; Make sure there's no garbage functions hanging around.
+    (loop while (not (zerop (length fs)))
+          do (vector-pop fs)
+          do (setf (aref fs (fill-pointer fs)) nil))))
```

Comment:
Why not to simply create a fresh vector?
Like this:
```lisp
(setf fs (make-array 0 :adjustable t :fill-pointer t))
```

Reply:
In my experience, SBCL won't necessarily garbage collect the elements of a complex data structure if they're not manually set to `nil`.

Reply:
I'm pretty sure that they would be collected alongside the vector though.